proof/refine/X64/Finalise_R.thy

```diff
@@ -1392,40 +1392,13 @@ lemma postCapDeletion_irq_handlers'[wp]:
    \<lbrace>\<lambda>rv. valid_irq_handlers'\<rbrace>"
   by (wpsimp simp: Retype_H.postCapDeletion_def X64_H.postCapDeletion_def)
 
-crunch deletedIRQHandler
-  for ioports'[wp]: valid_ioports'
-  (wp: valid_ioports_lift'')
-
+(* extra constraints on IO ports are not needed since there is no valid_ioports to preserve in
+   Refine *)
```

Comment:
Hmm, this comment sort of feels like it's only useful now while we still have the context that there used to be extra constraints. Do you think it will still make sense in the future when we've forgotten that `valid_ioports'` ever existed and had to be reasoned about in Refine?

Reply:
This comment exists for only one reason: someone doing a `git log` in future and seeing that the definition has changed to drop very sane looking IO port cap constraints, replacing them with the exact same thing as IRQ handlers, and going "WTF happened". The comment will then indicate that the reason this is possible is due to valid_ioports' being gone. Does that actually help?

Reply:
I think that this is almost a style decision, but for that case shouldn't the person doing the `git log` get that information from the commit message?
My preference is that comments should make sense with the current state of the repository, and I sort of feel like someone looking at this definition in the future won't be expecting extra constraints to be needed, and will then be confused by the comment unless they do do that `git log`.
It's hard to get into that future mindspace though, so I definitely might be wrong and won't argue further if you want to keep the comment in.

Reply:
I'll keep it then, if only to help jog my memory if/when this comes up in the future... because getting this formulation right took me nearly a day and it wasn't at all obvious why IO port cap constraints were no longer needed, but this thing was. I think I ended up trying every combination at some point.